app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -98,7 +98,7 @@ fi
 <% end -%>
 
 register_host() {
-  curl --silent --show-error --cacert $SSL_CA_CERT --request POST <%= @registration_url %> \
+  register_tmp=$(mktemp) && curl -o $register_tmp --silent --show-error --cacert $SSL_CA_CERT --request POST <%= @registration_url %> \
```

Comment:
I think we should move the `register_tmp=$(mktemp)` outside the `register_host` and `register_katello_host`, to the same place we define the `SSL_CA_CERT` file. Then we can clean it up in the `cleanup_and_exit` function.
We can then pass the value of `register_temp` as a parameter to both `register` functions.

I think it would be much cleaner.

Reply:
Also: we are mixing `&&` and `;`.
I think we can just write it as separate lines:
``` sh
curl && bash
EXIT_CODE=$?
#...
cleanup_and_exit $EXIT_CODE
```